File: C++/Level.2/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    sort(people.begin(), people.end());
    int start = 0, end = people.size() - 1;
    while(start <= end) {
        int temp = people[end];
        while(start != end && temp + people[start] <= limit) {
            temp += people[start];
            ++start;
        }
        ++answer;
        --end;
    }
    return answer;
}

File: JavaScript/Level.2/구명보트.js
function solution(people, limit) {
    people.sort((a, b) => {
        return a - b;
    });
    let answer = 0;
    let lt = 0,
        rt = people.length - 1;
    while (lt <= rt) {
        if (lt === rt) {
            ++answer;
            break;
        }
        if (people[lt] + people[rt] <= limit) {
            ++lt;
            --rt;
            ++answer;
        } else {
            --rt;
            ++answer;
        }
    }
    return answer;
}
